lib/pure/asyncdispatch.nim

```diff
@@ -1436,6 +1436,8 @@ else:
         if (customSet * events) != {}:
           isCustomEvent = true
           processCustomCallbacks(p, fd)
+          if events.contains(Event.Process):
+            p.selector.unregister(fd.int)
```

Comment:
Why do this here in runOnce?

Reply:
I put it there because that is where the call to processCustomCallbacks() that handles the "Event.Process" callback is made. And I confirmed that the leak in signalfd is also eliminated.
Would another location be more appropriate?

Reply:
I don't know but `runOnce` typically does not `unregister` anything so it seems off.

Reply:
@Araq 
It seems to be handled by modifying the asynctools and unregister() in the callback. 
My apologies for submitting the wrong pull request.
Since asynctools is not maintained, I would like to fork it and use it. 
Thank you for your time and effort in dealing with my simple mistake.